src/js_native_api_v8.cc

```diff
@@ -2997,3 +2997,38 @@ napi_status napi_get_instance_data(napi_env env,
 
   return napi_clear_last_error(env);
 }
+
+napi_status napi_arraybuffer_detach(napi_env env,
+                                     napi_value arraybuffer) {
+  CHECK_ENV(env);
+  CHECK_ARG(env, arraybuffer);
+
+  v8::Local<v8::Value> value = v8impl::V8LocalValueFromJsValue(arraybuffer);
+  if (!value->IsArrayBuffer()) {
+    napi_throw_type_error(
+        env,
+        "ERR_NAPI_INVALID_ARRAYBUFFER_ARGS",
```

Comment:
The errors should be added to https://github.com/nodejs/node/blob/master/lib/internal/errors.js.  And since the string is different, should likely be different erors as well. For example look at lines 1083 through 1086 as an example of other N-API messages.

Reply:
In `napi_get_arraybuffer_info()` we use
```C++
RETURN_STATUS_IF_FALSE(env, value->IsArrayBuffer(), napi_invalid_arg);
```
We should not throw here. We should return `napi_invalid_arg`.